### PR TITLE
test: migrated middleware.test.js from tap to node:test

### PR DESCRIPTION
**Pull request by Tony133**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Proposal

 - migrated `middleware.test.js` file from `tap` to `node:test` 🔥

### CODE REVIEW

File: test/middleware.test.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ test('Should be able to override the default use API', t => {
   t.plan(1)
   const fastify = Fastify()
   fastify.decorate('use', () => true)
-  t.equal(fastify.use(), true)
+  t.assert.equal(fastify.use(), true)
```
Comment (dancastillo) [inline test/middleware.test.js:13]: Can we use `strictEqual` here
```suggestion
  t.assert.strictEqual(fastify.use(), true)
```

File: test/middleware.test.js [hunk 2]
```diff
@@ -20,17 +20,16 @@ test('Cannot decorate use twice', t => {
...
 })
 
 test('Encapsulation works', t => {
-  t.plan(1)
   const fastify = Fastify()
 
   fastify.register((instance, opts, done) => {
     instance.decorate('use', () => true)
-    t.equal(instance.use(), true)
+    t.assert.equal(instance.use(), true)
```
Comment (climba03003) [inline test/middleware.test.js:32]:
```suggestion
    t.assert.strictEqual(instance.use(), true)
```

**Review by Uzlopak — APPROVED** [review]
LGTM

**Review by gurgunday — APPROVED** [review]
lgtm

**Pull request merged**